parser/src/parsercontext.cpp

```diff
@@ -76,6 +77,31 @@ ParserContext::ParserContext(
 
      // TODO: detect ADDED files
    });
+
+  // Fill moduleDirectories vector
+  if (options.count("modules")) {
+    const std::string& modulesFilePath = options["modules"].as<std::string>();
+
+    std::ifstream fileStream(modulesFilePath);
+
+    if (!fileStream.good()) {
+      LOG(error) << "Failed to open modules file: " << modulesFilePath;
+      exit(1);
+    }
+
+    LOG(info) << "Processing modules file: " << modulesFilePath;
+
+    std::string line;
+    while (std::getline(fileStream, line)) {
+      try {
+        const fs::path p = fs::canonical(line);
+        moduleDirectories.push_back(p.string());
+      } catch (...) {
+        LOG(error) << "Failed to process path from modules file: " << line;
+        exit(1);
```

Comment:
Same here, instead of an ungraceful exit, the improper line should be ignored. Also, maybe some information about the catched exception could be added to the log message, e.g. the exception message. Maybe try to catch a ` std::exception` first, and only keep the `catch (...)` as final handler.

Reply:
I think we can omit handling the out of memory issue for `push_back()` at this point, since we are preprocessing the input parameters here. If we run out of memory at this point, we can crash the app, we don't really have any solution for the problem in the first place.